docs/x509.rst

```diff
@@ -196,6 +196,24 @@ X.509 Certificate Object
             >>> isinstance(cert.signature_hash_algorithm, hashes.SHA256)
             True
 
+    .. attribute:: extensions
+
+        :type: :class:`collections.OrderedDict`
```

Comment:
Do we want to directly expose an OrderedDict, or do what we did for Names, with a custom type that's iterable, and has a get_by_oid method?